python/cudf/cudf/core/index.py

```diff
@@ -486,9 +486,15 @@ def append(self, other):
 
             if len(self) and len(other):
                 if is_mixed_with_object_dtype(this, other):
+                    got_dtype = list(
+                        set([this.dtype, other.dtype])
+                        - set([np.dtype("object")])
+                    )
```

Comment:
This seems like a lot of unnecessary conversions between lists and sets just to get the non `object` dtype. I think it would be easier and cleaner to read to just use an `if else` here:
```
got_dtype = other.dtype if this.dtype == np.dtype("object") else this.dtype
```

Reply:
Done, this is much better 🙏